src/ui/ResultFolding/ResultFolding.ts

```diff
@@ -302,51 +300,40 @@ export class ResultFolding extends Component {
     window.scrollTo(0, new Win(window).scrollY() + resultElem.getBoundingClientRect().top);
   }
 
-  private displayThoseResults(results: IQueryResult[]): Promise<boolean> {
+  private async displayThoseResults(results: IQueryResult[]): Promise<boolean> {
     const childResultsPromises = _.map(results, result => {
       return this.renderChildResult(result);
     });
 
-    return Promise.all(childResultsPromises).then((childsToAppend: HTMLElement[]) => {
-      $$(this.results).empty();
-      _.each(childsToAppend, oneChild => {
-        this.results.appendChild(oneChild);
-      });
-      return true;
+    const childsToAppend: HTMLElement[] = await Promise.all(childResultsPromises);
+    $$(this.results).empty();
+    _.each(childsToAppend, oneChild => {
```

Comment:
One thing I like to slowly change in every file whenever we fix something is to remove the `import * as _ from 'underscore'` and instead import individual method that we need.

This means we will be eventually able to remove some dead code using webpack.